src/Orleans/Logging/LoggerWrapper.cs

```diff
@@ -0,0 +1,182 @@
+            this.ipEndpoint = ipEndpoint;
+            this.loggerFactory = loggerFactory;
+        }
+
+        public override void Log(int errorCode, Severity sev, string format, object[] args, Exception exception)
+        {
+            var message = FormatLogMessage(DateTime.UtcNow, sev, this.name, FormatMessageText(format, args), this.ipEndpoint, exception, errorCode, true);
+            this.logger.Log<string>(SeverityToLogLevel(sev), new EventId(0), message, exception, (msg, exc) => $"{msg}. Exception:{exc}");
+        }
+
+        private static string FormatMessageText(string format, object[] args)
+        {
+            // avoids exceptions if format string contains braces in calls that were not
+            // designed to use format strings
+            return (args == null || args.Length == 0) ? format : String.Format(format, args);
+        }
+
+        private static string FormatLogMessage(
+         DateTime timestamp,
+         Severity severity,
+         string caller,
+         string message,
+         IPEndPoint myIPEndPoint,
+         Exception exception,
+         int errorCode,
+         bool includeStackTrace)
+        {
+            if (severity == Severity.Error)
+                message = "!!!!!!!!!! " + message;
+
+            string ip = myIPEndPoint == null ? String.Empty : myIPEndPoint.ToString();
+            string exc = includeStackTrace ? LogFormatter.PrintException(exception) : LogFormatter.PrintExceptionWithoutStackTrace(exception);
+            string msg = String.Format("[{0} {1,5}\t{2}\t{3}\t{4}\t{5}]\t{6}\t{7}",
+                LogFormatter.PrintDate(timestamp),           //0
+                Thread.CurrentThread.ManagedThreadId,   //1
+                severity.ToString(),    //2
+                errorCode,                              //3
+                caller,                                 //4
+                ip,                                     //5
+                message,                                //6
+                exc);      //7
+
+            return msg;
+        }
+
+        private static LogLevel SeverityToLogLevel(Severity severity)
+        {
+            switch (severity)
+            {
+                case Severity.Off: return LogLevel.None;
+                case Severity.Error: return LogLevel.Error;
+                case Severity.Warning: return LogLevel.Warning;
+                case Severity.Info: return LogLevel.Information;
+                case Severity.Verbose: return LogLevel.Debug;
+                default: return LogLevel.Trace;
+            }
+        }
+        private Severity FindSeverityForLogger(ILogger logger)
+        {
+            //traversal from the lowest LogLevel to the highest to find the Severity of current Logger
+            //If Trace is enabled, then minimun enabled LogLevel is Trace, which maps to Severity being Verbose2
+            if (logger.IsEnabled(LogLevel.Trace))
+                return Severity.Verbose2;
```

Comment:
Should this be Verbose3?

Reply:
yeah I think you are right, min LogLevel is Trace means enabling all loglevel, whose counterpart should be Verbose3